app/src/lib/git/diff.ts

```diff
@@ -110,14 +111,13 @@ export async function getCommitDiff(
     repository.path,
     'getCommitDiff'
   )
-  if (isBufferTooLarge(output)) {
-    return { kind: DiffType.TooLarge, length: output.length }
+
+  const largeDiff = buildLargeTextDiff(output)
+  if (largeDiff != null) {
+    return largeDiff
```

Comment:
There are 3 conditions that must be met for an `ILargeTextDiff` to be returned from `buildLargeTextDiff`. If none of the following are met, we return `null` because the passed in buffer doesn't meet the conditions to be an `ILargTextDiff` object.
1. `!didReadAllBytes || !isValidBuffer(buffer)`
1. `isBufferTooLarge(buffer)`
1. `isDiffTooLarge(diffText)`

> ...maybe bufferTooLargeCommitDiff, diffTooLargeCommitDiff, invalidBufferCommitDiff since the information we provide for each commit diff is unique to the error case but we don't care what's inside much beyond that.

The reason why the name is not _[noun]TooLarge_ or _invalid[noun]_ is because the words `too` and `invalid` imply, at least to me, that there is no further work we can do with the named object. By omitting any indication that something is too large or invalid allows me to reason more clearly about _how_ this object is to be used: as a way to defer opening a diff that could degrade the UX. 

> I prefer the isBufferTooLarge/etc. boolean checks that were there before...Are we switching to the buildLargeTextDiff method because we want to switch over to doing all checks in every case (buffer invalid, buffer too large, diff too large)?

The reason the function `buildLargeTextDiff` was extracted is because we were duplicating logic in the two functions `getCommitDiff` and `getWorkingDirectoryDiff`. 

> ...ex: maybe bufferTooLargeCommitDiff, diffTooLargeCommitDiff, invalidBufferCommitDiff...

Not sure I'm clear on this. Are you implying that we creating different interfaces for the three different conditions we check for?

If so would something like the following satisfy that requirement?
`type LargeTextDiff = 'bufferTooLarge' | 'diffTooLarge' | ILargeTextDiff`
Doing this would allow us to distinguish between a `ILargeTextDiff` that can be rendered and one that cannot (we can currently do this now, but by inspecting properties).

Reply:
> If none of the following are met, we return null because the passed in buffer doesn't meet the conditions to be an ILargTextDiff object.

Once you read the implementation of `buildLargeTextDiff`, it's clear we are checking against 3 test conditions and maybe returning the text diff object. I don't think returning an `object|null` is as intuitive as exposing one or two boolean checks UNLESS we need to check for all 3 cases in both `getCommitDiff` and `getWorkingDirectoryDiff` functions in the exact same way.

> The reason the function buildLargeTextDiff was extracted is because we were duplicating logic in the two functions getCommitDiff and getWorkingDirectoryDiff.

In this case, I'm not sure it's worth it since we have to do a check against null now in both places instead. The tradeoff is between clarity and duplication.

> Are you implying that we creating different interfaces for the three different conditions we check for?

more like we bring back the boolean checks but have a helper method that defines the output object. ex:
```
if (isDiffTooLarge(diffText)) {	
  return bufferTooLargeCommitDiff()	
}
```
we could even provide better boolean checks (re: your comment of `too` being unclear as a descriptor), ex: s/`isDiffTooLarge`/`diffExceedsMaxLineLength`